### PR TITLE
feature/SOF-3766

### PR DESCRIPTION
**Pull request by mohammadidinani**

### CODE REVIEW

File: src/components/ThreeDEditor.js [hunk 1]
```diff
@@ -339,8 +323,15 @@ export class ThreeDEditor extends React.Component {
         if (this.state.isThreejsEditorModalShown) {
             return <ThreejsEditorModal
                 show={this.state.isThreejsEditorModalShown}
-                onHide={this.ToggleThreejsEditorModal}
-                materials={[this.props.material]}
+                onHide={(material) => {
+                    this.setState({
```
Comment (timurbazhirov) [inline src/components/ThreeDEditor.js]: Let's move this to a handler function above

File: src/components/ThreeDEditor.js [hunk 2]
```diff
@@ -307,11 +292,10 @@ export class ThreeDEditor extends React.Component {
         return <WaveComponent
             ref={(el) => {this.WaveComponent = el}}
             triggerHandleResize={this.state.viewerTriggerResize}
-            structure={this.props.material}
+            structure={this.state.material}
             cell={this.unitCell}
-            name={this.props.material.name}
+            name={this.state.material.name}
```
Comment (timurbazhirov) [inline src/components/ThreeDEditor.js]: We should update the state material when props change

File: src/components/ThreeDEditor.js [hunk 3]
```diff
@@ -46,6 +46,8 @@ export class ThreeDEditor extends React.Component {
                 atomRadiiScale: 0.2,
                 atomRepetitions: 1,
             },
+            originalMaterial: this.props.material,
+            material: this.props.material.clone(),
```
Comment (timurbazhirov) [inline src/components/ThreeDEditor.js]: Let's add a comment explaining the logic for using originalMaterial and material

File: src/utils.js [hunk 4]
```diff
@@ -53,5 +98,21 @@ export function materialsToThreeDSceneData(materials) {
...
+    const ambientLight = new THREE.AmbientLight("#202020");
+    ambientLight.name = "Ambient - DO NOT MODIFY";
+    directionalLight.position.copy(new THREE.Vector3(0.2, 0.2, -1).normalize());
+    directionalLight.intensity = 1.2;
+    lightsGroup.add(directionalLight);
+    lightsGroup.add(ambientLight);
+    wave.scene.add(lightsGroup);
+    lightsGroup.position.set(-50, 0, 10);
+    wave.render();
+
```
Comment (timurbazhirov) [inline src/utils.js]: We need docstrings above

File: src/components/ThreeDEditor.js (hunk 3 above)
Comment (mohammadidinani) [reply src/components/ThreeDEditor.js]: Done.

File: src/components/ThreeDEditor.js (hunk 2 above)
Comment (mohammadidinani) [reply src/components/ThreeDEditor.js]: Done.

File: src/components/ThreeDEditor.js (hunk 1 above)
Comment (mohammadidinani) [reply src/components/ThreeDEditor.js]: Done.

File: src/utils.js (hunk 4 above)
Comment (mohammadidinani) [reply src/utils.js]: This part is removed in SOF-3632.

**Pull request merged**